aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1596,13 +1596,13 @@ Tensor& vdot_out(const Tensor& self, const Tensor& other, Tensor& result) {
   return result.fill_(self.vdot(other));
 }
 
-namespace {
 bool should_fold_into_mm(const Tensor& tensor1, const Tensor& tensor2) {
   auto dim_tensor1 = tensor1.dim();
   auto dim_tensor2 = tensor2.dim();
   if (dim_tensor1 >= 3 && (dim_tensor2 == 1 || dim_tensor2 == 2)) {
     if (dim_tensor1 == 3 && dim_tensor2 == 2 &&
-        tensor1.stride(-1) != 1 && tensor1.stride(0) == tensor1.size(1) * tensor1.size(2)) {
+        tensor1.strides().back() != 1 &&
+        tensor1.strides().front() == tensor1.sizes().cbegin()[1] * tensor1.sizes().cbegin()[2]) {
```

Comment:
If you're going to optimize like this, it's probably even better to fetch tensor.strides() only once and don't rely on the compiler to CSE it.